components/clarinet-format/src/formatter/mod.rs

```diff
@@ -376,36 +395,29 @@ impl<'a> Aggregator<'a> {
         acc.push(')');
         acc
     }
+
     fn format_map(&self, exprs: &[PreSymbolicExpression], previous_indentation: &str) -> String {
-        let mut acc = "(define-map ".to_string();
+        let func_type = self.display_pse(exprs.first().unwrap(), "");
+        let mut acc = format!("({func_type} ");
         let indentation = &self.settings.indentation.to_string();
-        let space = format!("{}{}", previous_indentation, indentation);
+        let space = format!("{}{}", indentation, previous_indentation);
+        acc.push_str(&self.format_source_exprs(slice::from_ref(&exprs[1]), previous_indentation));
+        let mut iter = exprs[2..].iter().peekable();
+        while let Some(expr) = iter.next() {
+            let trailing = get_trailing_comment(expr, &mut iter);
 
-        if let Some((name, args)) = name_and_args(exprs) {
```

Comment:
this was overcomplicated and error-prone. I replaced it with this boilerplate-y mess, but we can abstract this away later